### PR TITLE
Extend TooltipManager to support multiple types of tooltip

### PR DESCRIPTION
**Pull request by arlyon**
This PR extends the tooltip manager so that many implementations can be used.

- Creates the `Tooltip` interface for multiple tooltip types in the `TooltipManager`
- Creates a `TooltipComponent` interface that must be implemented by any tooltip-compatible component.
- Creates a new `PanelTooltip` that uses the `PanelComponent` (to be used in #725).

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipComponent.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	 * @see Client#getModIcons() Gets the list of mod icons from the client.
+	 */
+	void setModIcons(IndexedSprite[] modIcons);
+
+	/**
+	 * Sets the position of the RenderableEntity to the given Point.
+	 *
+	 * @param position The desired position.
+	 */
+	void setPosition(Point position);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipComponent.java]: This is not needed at all, each RenderableEntity have Point in render method, use that one.

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/Tooltip.java [hunk 2]
```diff
@@ -25,16 +25,33 @@
...
 {
-	private final String text;
-	private boolean followMouse = true;
-	private Point position = new Point();
+	/**
+	 * Creates the TooltipComponent for the tooltip.
+	 *
+	 * @return The TooltipComponent
+	 */
+	TooltipComponent getTooltipComponent();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/Tooltip.java]: Can you rename this method to not look like getter?

File: runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipComponent.java (hunk 1 above)
Comment (arlyon) [reply runelite-client/src/main/java/net/runelite/client/ui/overlay/tooltip/TooltipComponent.java]: Do you mind elaborating on that? I'm not sure how the tooltip overlay would assign a component a location without it.

**Comment by arlyon** [comment]
closing due to age. might rework it

**Pull request closed without merging**